test/e2e/framework.go

```diff
@@ -794,10 +794,18 @@ func (data *TestData) deployFlowVisibilityClickHouse() (string, error) {
 	}
 
 	// check clickhouse service http port for service connectivity
-	chSvc, err := data.GetService("flow-visibility", "clickhouse-clickhouse")
-	if err != nil {
-		return "", err
+	var chSvc *corev1.Service
+	if err := wait.PollImmediate(defaultInterval, defaultTimeout, func() (bool, error) {
```

Comment:
have you observed that the Service is not available right away after applying the yaml? This would be strange, especially given that we wait for the CH Pod to be ready.

Reply:
yes, I noticed the same thing happen in both antrea and theia e2e, after we upgrade clickhouse server and operator.

Reply:
Did something change in the ClickHouse operator? Maybe it now waits longer to create the Service (I assume the Service is created by the operator).

Reply:
hmmm, I looked around and didn't see any changes been mentioned about the Service creation. The Service is created by a custom resource `ClickHouseInstallation`. cc @yanjunz97 who has more experience on CH - do you just happen to know the possible reason?

Reply:
I'm not sure about the root cause, but I remember @yuntanghsu has observed the similar problem before we upgraded the ClickHouse at https://github.com/antrea-io/theia/pull/263

Reply:
This issue I encountered before is:

We are using env variable "CLICKHOUSE_URL"=clickhouse-clickhouse.flow-visibility.svc.cluster.local when connecting to clickhouse from theia-manager. However, kube-dns is unstable for few seconds when a new service is added. Adding retry can prevent theia-manager from directly sending error back to the user when ping is temporarily failed.

Not sure if it is the same issue?

Reply:
Sorry, please ignore my last comment. It's irrelevant. I just take another look at ClickHouse operator code. It seems they changed the order of service creation since v0.20.2. As stating in the commit here: https://github.com/Altinity/clickhouse-operator/commit/5d3ab52312eef0b807f35af72b1f65e9f722c7d5

Reply:
Thanks @yanjunz97 and @yuntanghsu . Will it make more sense if we check Service first then the Pod? As `reconcileCHIService` is called later than `reconcileHostStatefulSet`

Reply:
Yep, I don't see much difference and basically I prefer to keep FIFO

Reply:
the order is fine as it is. It makes sense to check the Pods first.
As long as we have an explanation, the new code looks fine to me